File: hw/ramos/u1/hw16Libraries/JavaScript/Main.js
import IncomeTax from "./incomeTax.js";

function main() {
  console.log("=== Income Tax Calculator ===\n");

  import('prompt-sync').then(({ default: promptSync }) => {
    const prompt = promptSync({ sigint: true });

    const salary = parseFloat(prompt("Enter your monthly salary (USD): "));
    const expenses = parseFloat(prompt("Enter your monthly deductible expenses (USD): "));

    const taxCalculator = new IncomeTax();

    const monthlyTax = taxCalculator.calculateMonthlyTax(salary, expenses);
    const annualTax = monthlyTax * 12;
    const annualIncome = salary * 12;
    const bracketInfo = taxCalculator.findTaxBracket(annualIncome - expenses * 12);

    console.log("\n-------------------------------------");
    console.log(bracketInfo);
    console.log(`Estimated monthly tax: $${monthlyTax}`);
    console.log(`Estimated annual tax:  $${annualTax}`);
    console.log("-------------------------------------");
  });
}

main();

File: hw/ramos/u1/hw16Libraries/Python/Main.py
from IncomeTax import IncomeTax

def main():
    print("=== Ecuador Income Tax Calculator (Year 2024) ===\n")

    salary = float(input("Enter your monthly salary (USD): "))
    expenses = float(input("Enter your monthly deductible expenses (USD): "))

    tax_calculator = IncomeTax()

    monthly_tax = tax_calculator.calculate_monthly_tax(salary, expenses)
    annual_tax = monthly_tax * 12

    # Determine tax bracket
    annual_income = salary * 12
    bracket_info = tax_calculator.find_tax_bracket(annual_income - expenses * 12)

    print("\n-------------------------------------")
    print(bracket_info)
    print(f"Estimated monthly tax: ${monthly_tax}")
    print(f"Estimated annual tax:  ${annual_tax}")
    print("-------------------------------------")

if __name__ == "__main__":
    main()

